ophyd/v2/core.py

```diff
@@ -510,9 +493,9 @@ def set_sim_put_proceeds(signal: Signal[T], proceeds: bool):
         event.clear()
 
 
-def monitor_sim_value(signal: Signal[T], callback: ReadingValueCallback[T]) -> Monitor:
+def set_sim_callback(signal: Signal[T], callback: ReadingValueCallback[T]) -> None:
     """Monitor the value of a signal that is in sim mode"""
-    return _sim_backends[signal].monitor_reading_value(callback)
+    return _sim_backends[signal].set_callback(callback)
```

Comment:
Hmm, I forgot we did this, this leads to a hard-to-debug failure mode:
```python
set_sim_callback(print)
sig.put(value)  # prints value
sig.stage()  # replaces the callback
sig.put(value)  # doesn't print value anymore
```
I think we should probably make the `set_callback` methods more strict so it errors if you try and set a callback and one is already set

Reply:
Could you expand this example? Am struggling to recreate it.

Should I make an unset_callback method to a SignalBackend?

Reply:
I think I've done this now for all use cases.